manifests/cli/config.pp

```diff
@@ -24,8 +24,10 @@
   validate_string($ssh_private_key_content)
 
   # required by PuppetX::Jenkins::Provider::Clihelper base
-  package { 'retries':
-    provider => 'gem',
+  if ! defined(Package['retries']) {
```

Comment:
I usually try to resolve "generic" package issues using ensure_packages as well. However it doesn't work in this specific case. The challenge here is that one needs to use a different provider for this package. The following results in a duplicate declaration because it cannot be merged properly:
```
#config.pp
ensure_packages(['retries'],{'provider' => 'gem'})
```
```
#other.pp
ensure_packages(['retries'],{'provider' => 'puppet_gem'})
```